extendr-api/src/wrapper/list.rs

```diff
@@ -88,12 +88,17 @@ impl List {
     ///     assert_eq!(names, vec!["a", "b"]);
     /// }
     /// ```
-    pub fn from_hashmap<K>(val: HashMap<K, Robj>) -> Result<Self>
+    pub fn from_hashmap<K, V>(val: HashMap<K, V>) -> Result<Self>
     where
+        V: IntoRobj,
```

Comment:
Should this be `TryFrom`?

Reply:
I don't think so. All Rust types can be made into R types, although not uniquely. So the conversion is not falliable. 

What I should have done is this for clarity

```diff
diff --git a/extendr-api/src/wrapper/list.rs b/extendr-api/src/wrapper/list.rs
index 4f3a0bbc7..87a53e476 100644
--- a/extendr-api/src/wrapper/list.rs
+++ b/extendr-api/src/wrapper/list.rs
@@ -90,13 +90,11 @@ impl List {
     /// ```
     pub fn from_hashmap<K, V>(val: HashMap<K, V>) -> Result<Self>
     where
-        V: IntoRobj,
+        V: Into<Robj>,
         K: Into<String>,
     {
-        let (names, values): (Vec<_>, Vec<_>) = val
-            .into_iter()
-            .map(|(k, v)| (k.into(), v.into_robj()))
-            .unzip();
+        let (names, values): (Vec<_>, Vec<_>) =
+            val.into_iter().map(|(k, v)| (k.into(), v.into())).unzip();
         let mut res: Self = Self::from_values(values);
         res.set_names(names)?;
         Ok(res)

```

Reply:
Can we make a follow up issue to trash this? I think it would be good. In a test case I'm about to push i have 

```rs

impl TryFrom<Robj> for Point {
    type Error = Error;

    fn try_from(value: Robj) -> std::result::Result<Self, Self::Error> {
        let inner_vec = Doubles::try_from(value)?;
        let x = inner_vec[0].inner();
        let y = inner_vec[1].inner();
        Ok(Point { x, y })
    }
}

impl From<Point> for Doubles {
    fn from(value: Point) -> Self {
        Doubles::from_values([value.x, value.y])
    }
}

impl From<Point> for Robj {
    fn from(value: Point) -> Self {
        Robj::from(Doubles::from(value))
    }
}
#[extendr]
fn test_hm_custom_try_from(mut x: HashMap<&str, Point>) -> List {
    x.insert("inserted_value", Point {x : 3.0, y: 0.1415});
    List::from_hashmap(x).unwrap()
}
```

which works quite well